app/api/users.py

```diff
@@ -208,6 +208,9 @@ def before_update_object(self, user, data, view_kwargs):
         #     data['thumbnail_image_url'] = uploaded_images['thumbnail_image_url']
         #     data['icon_image_url'] = uploaded_images['icon_image_url']
 
+        if has_access('is_admin') and data.get('deleted_at') != user.deleted_at:
```

Comment:
```python
if has_access('is_admin') and data.get('deleted_at')  and data['deleted_at']!= user.deleted_at:```

Reply:
@CosmicCoder96 when we used this condition then the restoring event is not working because `data.get('deleted_at')` is null while restoring. I'm opening a follow up PR for this. I tested by removing it and it is working fine